tests/test_protocols/test_generator.py

```diff
@@ -86,11 +91,104 @@ def _start_oef_node(self, network_node):
     def setup_class(cls):
         """Set the test up."""
         cls.runner = CliRunner()
-        cls.agent_name = "agent_1"
         cls.cwd = os.getcwd()
         cls.t = tempfile.mkdtemp()
         os.chdir(cls.t)
 
+    def test_compare_latest_generator_output_with_test_protocol(self):
+        """Test that the "t_protocol" test protocol matches with what the latest generator generates based on the specification."""
+        # check protoc is installed
+        res = shutil.which("protoc")
```

Comment:
@5A11 why are you not just calling the cli command to generate the protocol? It would save you code duplication. The below will easily diverge from the generator command logic and is therefore not recommended.

Reply:
If I do it using generate cli command, then I cannot supply the `path_to_protocol_package` argument so the generated protocol package would not quite match the actual test protocol due to import path differences.

Unless I open up that argument to the generate cli command as well, which means whenever you want to generate, you have to also give it the `path_to_protocol_package` argument which does not really makes sense and is not at all a clean solution.

Reply:
Making the `path_to_protocol_package` an optional argument in the cli `aea generate protocol` command solves the problem.